server/src/main/java/org/opensearch/wlm/ResourceType.java

```diff
@@ -61,20 +75,30 @@ public String getName() {
         return name;
     }
 
-    /**
-     * Gets the resource usage for a given resource type and task.
-     *
-     * @param task the task for which to calculate resource usage
-     * @return the resource usage
-     */
-    public long getResourceUsage(Task task) {
-        return getResourceUsage.apply(task);
-    }
-
     public boolean hasStatsEnabled() {
         return statsEnabled;
     }
 
+    public double calculateQueryGroupUsage(List<QueryGroupTask> tasks, Supplier<Long> nanoTimeSupplier) {
+        return resourceUsageCalculator.calculateResourceUsage(tasks, nanoTimeSupplier);
+    }
+
+    public double calculateTaskUsage(QueryGroupTask task, Supplier<Long> nanoTimeSupplier) {
+        return resourceUsageCalculator.calculateTaskResourceUsage(task, nanoTimeSupplier);
```

Comment:
Currently `Task` does not maintain running time it carries the `startTime` and `startTimeNanos` in order to capture current time to determine the total running we will need to rely on the current time.
Given SearchBackpressure also uses `System::nanoTime` and the number of tasks in the system will barely be in order of `100 `s.

Given the nature of how `System::nanoTime` works (It performs I/O so no magic dust here). On *nix platforms it is close to 25 +- 5 ns but on windows it could be in microseconds.